### DISCUSSION TITLE
Setting value with a style breakes textarea

### DISCUSSION THREAD

**Discussion by ipsavitsky** [charmbracelet/bubbles#708] [Q&A] [answered]
**Describe the bug**

I'm trying to set the textarea value to a string that has styles rendered. I expect the output to override the default styles, instead the textarea outputs the string escaped

**Setup**
Please complete the following information along with version numbers, if applicable.
 - OS Ubuntu on WSL
 - Shell bash
 - Terminal Emulator Windows Terminal
 - Terminal Multiplexer zellij
 - Locale en_US.UTF-8

**To Reproduce**
- Compile the source in **Source Code section**
- Start the compiled binary
- Press arrow up

**Source Code**
```go
package main

import (
	// "fmt"
	"log"

	"github.com/charmbracelet/bubbles/textarea"
	tea "github.com/charmbracelet/bubbletea"
	"github.com/charmbracelet/lipgloss"
)

var textStyle = lipgloss.NewStyle().Padding(1, 2).Faint(true);

func main() {
	p := tea.NewProgram(initialModel(), tea.WithAltScreen())

	if _, err := p.Run(); err != nil {
		log.Fatal(err)
	}
}

type model struct {
	textarea textarea.Model
}

func initialModel() model {
	ti := textarea.New()
	ti.CursorStart()
	ti.Focus()

	return model{
		textarea: ti,
	}
}

func (m model) Init() tea.Cmd {
	return textarea.Blink
}

func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
	var cmds []tea.Cmd
	var cmd tea.Cmd

	switch msg := msg.(type) {
	case tea.KeyMsg:
		switch msg.Type {
		case tea.KeyEsc:
			if m.textarea.Focused() {
				m.textarea.Blur()
			}
		case tea.KeyCtrlC:
			return m, tea.Quit
		case tea.KeyUp:
			m.textarea.SetValue(textStyle.Render("Hello world"))
		default:
			if !m.textarea.Focused() {
				cmd = m.textarea.Focus()
				cmds = append(cmds, cmd)
			}
		}
	}

	m.textarea, cmd = m.textarea.Update(msg)
	cmds = append(cmds, cmd)
	return m, tea.Batch(cmds...)
}

func (m model) View() string {
	return m.textarea.View()
}
```

**Expected behavior**
After pressing arrow up a string "Hello world" appears in faint text.

**Screenshots**
![image](https://github.com/user-attachments/assets/0ac6c855-d277-44ee-b29f-d5ff580676bb)

**Additional context**
I am planning to render text with multiple styles applied to different parts and since I can't set multiple styles to textarea, I have opted for rerendering text on every event. If I am misusing the textarea component or any other part of the library, let me know.

**Comment by prithvijj** [comment]
Just adding some random notes
![image](https://github.com/user-attachments/assets/73d8f0cc-572b-4b37-a94b-6754eff89990)

After commenting out the code at https://github.com/charmbracelet/bubbles/blob/60796500b2e255dcb9d68b916a6fc5aa49cf5b47/runeutil/runeutil.go#L88 , and then trying it, the control sequence characters end up rendering things, but it causes weird behaviour obivously like

(pressing the Left Arrow key multiple times, causes the part of the text to be in `red` and in regular font color `white` )
![image](https://github.com/user-attachments/assets/da16c6dd-bad9-42ff-939e-ddcf24270307)

---

And there's the ability to set some styles within the `BlurredStyle` and `FocusedStyle` fields

![image](https://github.com/user-attachments/assets/03b65982-7d64-4e7e-a0a7-4aaf5f68fe08)

**Comment by bashbunni** [comment, marked as answer]
Hey @ipsavitsky textarea doesn't support stylized text. It's meant to just be plain text. If you wanted to display some text and modify it, I would recommend using a viewport to display the text, then use textarea to edit the unstyled/plain text.

You could switch between the two in your program if you wanted. 

Please let me know if that clarifies things here. If it's still unclear, feel free to ping me. I'm going to convert this to a discussion instead of an issue as it's not something we're planning to address as of right now.

**Reply by ipsavitsky** [reply]
I have rewritten the functionality I need myself, if this is not the intended usage of textarea - this is perfectly fine!
